### ISSUE TITLE
Grid's Should AutoReflow based on container size

### ISSUE THREAD

**Issue by SavannahBourgeois** [microsoft/fluentui#13574]
### Environment Information

- **Package version(s)**: all
- **Browser and OS versions**: all

### Describe the issue:

With WCAG2.1 updates, there is a new rule that requires all surfaces to have at most one scroll bar when at a screen size of 320x256 CSS pixels (added link to W3C documentation at the end). This means for any of our galleries in Office Online that use the underlying Grid component in fabric, we specify the columns and fabric adds those as expected. However, on smaller screens sometimes the columns are too wide and cause 2 directional scrolling which is a violation of this Reflow rule. Instead of making all of our individual galleries check for screen size before specifying the number of columns, it would be great to add a container/bounding rect for the grid component and have it auto update the number of columns based on what can fit.

### Please provide a reproduction of the issue in a codepen:

Using the swatch colorpicker example on the demo page (https://developer.microsoft.com/en-us/fluentui#/controls/web/swatchcolorpicker) open the devtools or resize the window to be quite small. 

The screenshot I've attached may be a little extreme to show the point, but it does happen for some of the galleries that use the Grid component at 200% zoom in Office Online. I'm happy to provide the examples in Office Online if that is helpful.

#### Actual behavior:
You will see the items get cut off and they are not accessible.

![itemscutoff](https://user-images.githubusercontent.com/42548121/84444299-4f5cae80-abf6-11ea-934c-9d4d410d3e59.PNG)

#### Expected behavior:
The grid will reflow the number of columns to be smaller so there are more rows and all items fit on the screen or bounding container.

### Documentation describing expected behavior

https://www.w3.org/WAI/WCAG21/Understanding/reflow.html

**Comment by SavannahBourgeois** [comment]
@makopch-ms @jspurlin as FYI

**Comment by paulgildea** [comment]
@JustSlone @dzearing - Looking through the code owners folder I don't see anyone assigned to the Grid component. Who would be a good candidate for addressing this issue?

**Comment by SavannahBourgeois** [comment]
@paulgildea @JustSlone @dzearing Can you help find an owner for this? We have a few accessibility bugs on it and need to know if it will be fixed in time or if we will need to do a workaround on our side.

**Comment by JustSlone** [comment]
Looking into this more, we're really talking about SwatchColorPicker not a Grid component per say. @ecraig12345 is code owner of SwatchColorPicker, tagging her in case she has any additional insight. 

So, this example given does actually meet the requirement of being able to fit in 320 pixels wide. 

I think the short term recommendation here would be to ensure that the specifications passed to the grid do not exceed 320 pixels. The calculation appears to be (cellWidth+10) * columnCount = total width. So in the example case it is 35+10 * 5 = 225 pixels wide, that should be small enough to fit 320 wide without a lot of trouble. Is there a specific scenario where this suggestion isn't possible?

Also, we can certainly look at updating the Grid used by SwatchColorPicker to be responsive in the future, but it's not currently in the list of components we are focusing on, so it would be later.

**Comment by ecraig12345** [comment]
This is basically a duplicate of #11481 as far as the functionality requested. As mentioned there, it's not feasible to add reflow in the current implementation which (incorrectly) uses a table, but could definitely be possible if we rewrite it to get rid of the table and use `<Stack wrap>` or something like that. Since we're working on a major version bump, we could consider that option.

**Comment by SavannahBourgeois** [comment]
So essentially we will need to calculate the number of columns to pass to a Grid for now until V8 which will (maybe?) have the Grid use a stack? We are using the Fabric Grid right now for our galleries like in the screenshot below, so it is not just for the Swatch Picker. 

@makopch-ms as FYI

![image](https://user-images.githubusercontent.com/42548121/85465080-368cbb00-b55d-11ea-8528-32077e83cb41.png)

**Comment by ecraig12345** [comment]
Unfortunately yeah. If you want to ensure that wrapping is supported in version 8, we'd be happy to accept a contribution converting it from a table to a Stack or some other setup. (I'm not 100% sure offhand that Stack would be the right solution, but it's the first obvious thing that comes to mind.)

It's also interesting that you have another use case for Grid--I hadn't expected it to be used anywhere outside SwatchColorPicker due to what appears to be extremely tight coupling with SwatchColorPicker, and had actually wanted to remove it as a separate component for that reason (#13124). If we do need to keep it around, we should at least rename it to ButtonGrid or something so that the name Grid is freed up for an abstraction of CSS grid (another thing we've wanted to add for a long time but haven't had time).
